src/appleseed/foundation/meta/tests/test_analysis.cpp

```diff
@@ -119,4 +119,44 @@ TEST_SUITE(Foundation_Image_Analysis)
 
         EXPECT_FEQ(1.0, rmsd);
     }
+
+    TEST_CASE(ComputeRMSDeviation_GivenImagesOfDifferentChannelSizes_ReturnsOne)
+    {
+        Image image1(4, 4, 2, 2, 3, PixelFormatFloat);
+        Image image2(4, 4, 2, 2, 4, PixelFormatFloat);
+
+        image1.clear(Color3f(0.0f));
+        image2.clear(Color4f(1.0f));
+
+        const double rmsd = compute_rms_deviation(image1, image2);
+
+        EXPECT_FEQ(1.0, rmsd);
+    }
+
+    TEST_CASE(ComputeRMSDeviation_GivenImagesOfDifferentTileSizes_ReturnsOne)
+    {
+        Image image1(4, 4, 1, 1, 4, PixelFormatFloat);
+        Image image2(4, 4, 2, 2, 4, PixelFormatFloat);
+
+        image1.clear(Color4f(0.0f));
+        image2.clear(Color4f(1.0f));
+
+        const double rmsd = compute_rms_deviation(image1, image2);
+
+        EXPECT_FEQ(1.0, rmsd);
+    }
+
+    TEST_CASE(ComputeRMSDeviation_GivenRedImageAndBlueImage_ReturnsAlmostRootTwoThirds)
```

Comment:
😂

Reply:
Oddly enough `FEQ(sqrt(2.0f / 2.0f), ...)` passed on most platforms but failed on lower `gcc` versions.

Reply:
Actually, the correct way to do this is

    EXPECT_FEQ(sqrt(2.0f / 3.0f), rmsd);

or

    EXPECT_FEQ_EPS(sqrt(2.0f / 3.0f), rmsd, 1.0e-6f);

(using a reasonable epsilon value)

Reply:
Ah, right, so I had actually tried the first way originally (which failed).

Neat, the second way is much better, updated.